src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -446,24 +446,32 @@ public override void Abort()
             Dispose();
         }
 
-        public override async Task CloseAsync(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
+        public override Task CloseAsync(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
         {
             _writeBuffer = null;
             ThrowIfNotConnected();
-            await CloseAsyncCore(closeStatus, statusDescription, cancellationToken).ConfigureAwait(continueOnCapturedContext: true);
-        }
-
-        private async Task CloseAsyncCore(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
-        {
-            ThrowOnInvalidState(State, WebSocketState.Open, WebSocketState.CloseReceived, WebSocketState.CloseSent);
 
             WebSocketValidate.ValidateCloseStatus(closeStatus, statusDescription);
 
+            try
+            {
+                ThrowOnInvalidState(State, WebSocketState.Open, WebSocketState.CloseReceived, WebSocketState.CloseSent);
+            }
+            catch (Exception exc)
+            {
+                return Task.FromException(exc);
+            }
+
+            return CloseAsyncCore(closeStatus, statusDescription, cancellationToken);
+        }
+
+        private Task CloseAsyncCore(WebSocketCloseStatus closeStatus, string? statusDescription, CancellationToken cancellationToken)
+        {
             _tcsClose = new TaskCompletionSource();
             _innerWebSocketCloseStatus = closeStatus;
             _innerWebSocketCloseStatusDescription = statusDescription;
             _innerWebSocket!.Invoke("close", (int)closeStatus, statusDescription);
```

Comment:
Can this Invoke call throw?

Reply:
Yes, that is what is being captured here: https://github.com/dotnet/runtime/issues/45538

Reply:
If this Invoke call can throw, then you need a try/catch that will catch that exception and store it into a task to be returned.

Reply:
perfect thanks

Reply:
Addressed in recent commits